### DISCUSSION TITLE
Index Into Vector

### DISCUSSION THREAD

**Discussion by DamynChipman** [idaholab/moose#19380] [Q&A General] [answered]
How do I index into a vector value such as `ADRealVectorValue`? For example, I need the x- and y-components of `_normal` for a class derived from `FVFluxKernel`.

Hoping this is a quick question, I can't find the solution in tutorials or documentation. Thanks.

**Comment by GiudGiud** [comment, marked as answer]
Hello

iirc, something like that:
```
ADRealVectorValue u;
auto u_x = u(0);
```

Guillaume
